src/services/ghost/GhostInlineCompletionProvider.ts

```diff
@@ -0,0 +1,892 @@
+import * as vscode from "vscode"
+import { GhostSuggestionsState } from "./GhostSuggestions"
+import { GhostSuggestionEditOperation } from "./types"
+import { GhostServiceSettings } from "@roo-code/types"
+
+// Constants
+const PLACEHOLDER_TEXT = "<<<AUTOCOMPLETE_HERE>>>"
+const MAX_CURSOR_DISTANCE = 5
+const COMMON_PREFIX_THRESHOLD = 0.8
+
+/**
+ * Inline Completion Provider for Ghost Code Suggestions
+ *
+ * Provides ghost text completions at the cursor position based on
+ * the currently selected suggestion group using VS Code's native
+ * inline completion API.
+ */
+export class GhostInlineCompletionProvider implements vscode.InlineCompletionItemProvider {
+	private suggestions: GhostSuggestionsState
+	private onIntelliSenseDetected?: () => void
+	private settings: GhostServiceSettings | null = null
+
+	constructor(
+		suggestions: GhostSuggestionsState,
+		onIntelliSenseDetected?: () => void,
+		settings?: GhostServiceSettings | null,
+	) {
+		this.suggestions = suggestions
+		this.onIntelliSenseDetected = onIntelliSenseDetected
+		this.settings = settings || null
+	}
+
+	/**
+	 * Update the suggestions reference
+	 */
+	public updateSuggestions(suggestions: GhostSuggestionsState): void {
+		this.suggestions = suggestions
+	}
+
+	/**
+	 * Update the settings reference
+	 */
+	public updateSettings(settings: GhostServiceSettings | null): void {
+		this.settings = settings
+	}
+
+	/**
+	 * Extract and join content from operations
+	 */
+	private extractContent(operations: GhostSuggestionEditOperation[], type?: "+" | "-"): string {
+		const filtered = type ? operations.filter((op) => op.type === type) : operations
+		return filtered
+			.sort((a, b) => a.line - b.line)
+			.map((op) => op.content)
+			.join("\n")
+	}
+
+	/**
+	 * Check if content is placeholder-only
+	 */
+	private isPlaceholderContent(content: string): boolean {
+		return content.trim() === PLACEHOLDER_TEXT
+	}
+
+	/**
+	 * Check if a deletion group is placeholder-only
+	 */
+	private isPlaceholderOnlyDeletion(group: GhostSuggestionEditOperation[]): boolean {
+		const deletedContent = this.extractContent(group, "-")
+		return this.isPlaceholderContent(deletedContent)
+	}
+
+	/**
+	 * Find common prefix between two strings
+	 */
+	private findCommonPrefix(str1: string, str2: string): string {
+		let i = 0
+		while (i < str1.length && i < str2.length && str1[i] === str2[i]) {
+			i++
+		}
+		return str1.substring(0, i)
+	}
+
+	/**
+	 * Check if added content has common prefix with deleted content
+	 */
+	private hasCommonPrefix(deletedContent: string, addedContent: string): boolean {
```

Comment:
name is slightly misleading, it is isPrefix; but tbh I think inlining will make it clearer

Reply:
Yes, let me change that.